src/pycsou/abc/operator.py

```diff
@@ -2049,25 +2052,30 @@ def from_sciop(cls, scipy_operator) -> "LinOp":
         This  is a simplified example for illustration puposes only. It may not abide by all the rules listed in the
         :ref:`developer-notes`.
 
+        Notes
+        -----
+        Scipy linear operators by default only accept 2D inputs only. In order to use N-D arrays,
+        :py:meth:`~pycsou.abc.operator.Map.from_sciop` reshapes the first N-1 stacking dimensions of the input array
+        into a single stacking dimension. The output shape is matched to the input's stacking dimensions.
+
         See Also
         --------
-        :py: meth:`~pycsou.abc.operator.LinOp.from_array`, :py: meth:`~pycsou.abc.operator.Map.from_source`
+        :py:meth:`~pycsou.abc.operator.LinOp.from_array`, :py:meth:`~pycsou.abc.operator.Map.from_source` and
+        :py:meth:`~pycsou.abc.operator.LinOp.to_scipy_operator`.
         """
 
-        def multidim_to_2d(func, arr):
-            return func(arr.transpose().reshape(arr.shape[-1], -1)).reshape(-1, arr.shape[-2::-1]).transpose()
+        if scipy_operator.dtype not in [elem.value for elem in pycrt.Width]:
+            warnings.warn("Computation may not be performed at the requested precision.", UserWarning)
 
-        def apply(cls, arr: pyct.NDArray) -> pyct.NDArray:
-            if arr.ndim <= 2:
-                return scipy_operator.matmat(arr.transpose()).transpose()
-            else:
-                return multidim_to_2d(scipy_operator.matmat, arr)
+        @pycrt.enforce_precision(i="arr")
+        def apply(self, arr: pyct.NDArray) -> pyct.NDArray:
+            out_shape = arr.shape[:-1] + (-1,) if arr.ndim > 1 else (-1,)
+            return scipy_operator.matmat(arr.reshape(-1, arr.shape[-1]).transpose()).transpose().reshape(*out_shape)
```

Comment:
Yes you are right, my initial suggestion was flawed as the dimension of the last axis could be changed by the operator.